presto-main/src/main/java/com/facebook/presto/type/DecimalToDecimalCasts.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.metadata.ParametricFunction;
+import com.facebook.presto.metadata.ParametricFunctionBuilder.ExtraParametersFunction;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.LongDecimalType;
+import com.facebook.presto.spi.type.ShortDecimalType;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.RoundingMode;
+
+import static com.facebook.presto.metadata.OperatorType.CAST;
+import static com.facebook.presto.metadata.Signature.comparableWithVariadicBound;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_CAST_ARGUMENT;
+import static com.facebook.presto.spi.type.LongDecimalType.unscaledValueToBigInteger;
+import static com.facebook.presto.spi.type.LongDecimalType.unscaledValueToSlice;
+import static com.facebook.presto.spi.type.StandardTypes.DECIMAL;
+import static com.facebook.presto.util.DecimalUtils.TEN_TO_NTH_SHORT;
+import static com.facebook.presto.util.DecimalUtils.doesFitPrecision;
+import static java.lang.String.format;
+
+public final class DecimalToDecimalCasts
+{
+    public static final Signature SIGNATURE = Signature.builder()
+            .operatorType(CAST)
+            .typeParameters(comparableWithVariadicBound("X", DECIMAL), comparableWithVariadicBound("Y", DECIMAL))
+            .argumentTypes("X")
+            .returnType("Y")
+            .build();
+
+    public static final ParametricFunction DECIMAL_TO_DECIMAL_CAST = ParametricFunction.builder(DecimalToDecimalCasts.class)
+            .signature(SIGNATURE)
+            .methods("shortToShortCast")
+            .extraParameters(shortToShortExtraParameters())
```

Comment:
do not call method which retuns lambda.
Instead pass method itself with `::` syntax.

```
.extraParameters(::shortToShortExtraParameters)
```